SourceCode/raytracer/raytracer/CRTTypes.h

```diff
@@ -284,28 +293,29 @@ class Matrix3x3 {
         float cosv = cos(rad);
         float sinv = sin(rad);
 
-        return Matrix3x3 {{
+        return Matrix3x3{ {
         cosv, -sinv, 0,
         sinv, cosv, 0,
         0,0, 1
-        }};
+        } };
     }
 
 
 
-    void toString() const {
+    std::string toString() const {
         std::string result = "";
-        for (int row = 0; row < 3; ++row) {
-            for (int col = 0; col < 3; ++col) {
+        for (size_t row = 0; row < 3; ++row) {
+            for (size_t col = 0; col < 3; ++col) {
                 result += std::to_string((*this)(row, col));
                 result += " ";
             }
             result += "\n";
         }
+        return result;
     }
 
 private:
-    float data[9];
+    std::array<float, 9> data;
```

Comment:
Be very carefull with std in such delicate places, we've seen various problems with it on different platforms

Reply:
@VladislavVulchevChaos  What do you mean? What bad could std::array do here?